core/profile.cpp

```diff
@@ -677,31 +677,28 @@ static void setup_gas_sensor_pressure(const struct dive *dive, const struct dive
 	std::vector<int> first(num_cyl, 0);
 	std::vector<int> last(num_cyl, INT_MAX);
 
-	int prev = dive->explicit_first_cylinder(dc);
-	prev = prev >= 0 ? prev : 0;
-	seen[prev] = 1;
-
-	event_loop loop("gaschange");
-	while (auto ev = loop.next(*dc)) {
-		int cyl = ev->gas.index;
-		int sec = ev->time.seconds;
+	int prev = -1;
+	gasmix_loop loop(*dive, *dc);
+	while (loop.has_next()) {
+		std::pair<int, int> gasmix_event = loop.next_cylinder_index();
```

Comment:
Here, and elsewhere: It could make sense to use structured bindings to make the meaning of this code more clear.

Reply:
Ha, I didn't know that they exist (yet) in C++ - done, looks neat.

Reply:
Yes this makes code more readable. Just be careful: these aren't real variables, but some obscure notion of "bindings", which can have weird behavior.